mpisppy/phbase.py

```diff
@@ -810,17 +811,10 @@ def _vb(msg):
 
         self._update_E1()  # Apologies for doing this after the solves...
         if (abs(1 - self.E1) > self.E1_tolerance):
-            if self.cylinder_rank == 0:
-                print("ERROR")
-                print("Total probability of scenarios was ", self.E1)
-                print("E1_tolerance = ", self.E1_tolerance)
-            quit()
+            raise RuntimeError(f"Total probability of scenarios was {self.E1};  E1_tolerance = ", self.E1_tolerance)
         feasP = self.feas_prob()
         if feasP != self.E1:
-            if self.cylinder_rank == 0:
-                print("ERROR")
-                print("Infeasibility detected; E_feas, E1=", feasP, self.E1)
-            quit()
+            raise RuntimeError(f"Infeasibility detected; E_feas={feasp}, E1={self.E1}")
```

Comment:
Same comment as above here (why not retain the cylinder rank 0 check).

Reply:
It is not clear to me whether the rank 0 check (which was my idea in this case) was a good idea. I think you want every rank to shut down if infeasibility is detected rather than relying on the user to have done -m mpy4py so it will shut everyone down